apis/core/v1alpha2/readinessprovider_types.go

```diff
@@ -0,0 +1,78 @@
+// Copyright YEAR VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ReadinessProviderSpec defines the desired state of ReadinessProvider
+type ReadinessProviderSpec struct {
+	CheckRef string `json:"checkRef"`
+
+	//+kubebuilder:validation:Optional
+	Repeatable bool `json:"repeatable"`
+
+	//+kubebuilder:validation:Optional
+	RepeatIntervalInSeconds int32                        `json:"repeatIntervalInSeconds"`
+	Conditions              []ReadinessProviderCondition `json:"conditions"`
+}
+
+type ReadinessProviderCondition struct {
+	Name              string                     `json:"name"`
+	ResourceExistence ResourceExistenceCondition `json:"resourceExistence"`
```

Comment:
Rename field to match struct name `ResourceExistenceCondition`.

Reply:
Done